ocaml/xapi/repository.ml

```diff
@@ -737,40 +737,29 @@ let apply_updates' ~__context ~host ~updates_info ~livepatches ~acc_rpm_updates
       ) ;
   (* Evaluate recommended/pending guidances *)
   let recommended_guidances, pending_guidances =
-    let new_recommended_gs =
+    let guidances =
       (* EvacuateHost will be applied before applying updates *)
       eval_guidances ~updates_info ~updates:acc_rpm_updates ~kind:Recommended
         ~livepatches:successful_livepatches ~failed_livepatches
       |> List.filter (fun g -> g <> Guidance.EvacuateHost)
     in
-    let recommended_guidances' =
+    let guidances' =
       merge_with_unapplied_guidances ~__context ~host ~kind:Recommended
-        ~guidances:new_recommended_gs
+        ~guidances
     in
-
-    let new_pending_gs =
-      eval_guidances ~updates_info ~updates:acc_rpm_updates ~kind:Absolute
-        ~livepatches:[] ~failed_livepatches:[]
-      |> List.filter (fun g -> not (List.mem g recommended_guidances'))
-    in
-    let pending_guidances' =
-      merge_with_unapplied_guidances ~__context ~host ~kind:Absolute
-        ~guidances:new_pending_gs
-    in
-
     match failed_livepatches with
     | [] ->
         (* No livepatch should be applicable now *)
         Db.Host.remove_pending_guidances ~__context ~self:host
           ~value:`reboot_host_on_livepatch_failure ;
-        (recommended_guidances', pending_guidances')
+        (guidances', guidances')
     | _ :: _ ->
         (* There is(are) livepatch failure(s):
          * the host should not be rebooted, and
          * an extra pending guidance 'RebootHostOnLivePatchFailure' should be set.
          *)
-        ( List.filter (fun g -> g <> Guidance.RebootHost) recommended_guidances'
-        , Guidance.RebootHostOnLivePatchFailure :: pending_guidances'
+        ( List.filter (fun g -> g <> Guidance.RebootHost) guidances'
```

Comment:
Do we need to expose `RebootHost` guidance to user when there is(are) livepatch failure(s)? If user need to tell if a pending “reboot host” is introduced by both "failed livepatch" and "some update", then we do not need to filter it out from the guidance here. 
@kc284 @alexbrett what is your thought?